doc/release.md

```diff
@@ -0,0 +1,103 @@
+   backward-compatible manner.
+3. `PATCH` version when there are backward-compatible bug fixes.
+4. `EXTRAVERSION` release-candidate suffix (e.g. `rc1`).
+
+We add pre-release tags using the format `MAJOR.MINOR.PATCH-EXTRAVERSION`.
+
+> [!NOTE]
+> Current firmware bundle files include a fourth version field for experimental builds. For
+> official releases, the fourth version field is always zero.
+
+## Release candidates
+
+Before each release, tags are made (see below) for at least one release candidate (`a.b.c-rc1`,
+followed by `a.b.c-rc2` and the subsequent release candidates, followed by the official `a.b.c`
+release). The intent is to freeze the code and allow testing.
+
+During the time between the `rc1` and the final release, the only changes that should be merged into
+the `main` branch are those to fix bugs found in the release candidate, to correct documentation,
+or to make other cosmetic fixes.
+
+## Release Procedure
+
+The following steps are required to be followed by firmware release engineers when creating a new
+`tt-zephyr-platforms` release.
+
+### Release Checklist
+
+Refer to the previous release checklist for steps required before tagging release candidates and
+final releases.
+
+### Tagging
+
+> [!NOTE]
+> This section uses a fake release version, `v1.2.3`, or `v1.2.3-rc1` as an example. Replease with
+> the appropriate release candidate or final release version.
+
+1. Update the version variables in the bundle version file `VERSION` file located in the root of
+    the Git repository to match the version for this release. The `EXTRAVERSION` variable is used
+    to identify release candidates. It is left empty for final releases.
+
+    ```
+    EXTRAVERSION = rc1
+    ```
+
+1. Post a PR with the updated `VERSION` file using `release: 1.2.3-rc1` or `release: 1.2.3` as
+    the commit subject. Merge the PR after CI is successful.
+
+2. Tag and push the version, using an annotated tag:
+
+    ```shell
+    git tag -s -m "tt-zephyr-platforms 1.2.3-rc1" v1.2.3-rc1
+    ```
+
+3. Verify that the tag has been
+    [signed correctly](https://docs.github.com/en/authentication/managing-commit-signature-verification/telling-git-about-your-signing-key),
+    `git show` for the tag must contain a signature (look for the `BEGIN PGP SIGNATURE` or
+    `BEGIN SSH SIGNATURE` marker in the output):
+
+    ```shell
+    git show v1.2.3-rc1
+    ```
+
+4. Push the tag:
+
+    ```shell
+    git push git@github.com:tenstorrent/tt-zephyr-platforms.git v1.2.3-rc1
+    ```
+
+Lastly, for final releases,
+
+6. Find the generated
+    [draft release](https://github.com/tenstorrent/tt-zephyr-platforms/releases), edit the release
+    notes, and publish the release.
+
+7. Announce the release via official Tenstorrent channels and provide a link to the
+    GitHub release page.
+
+### Publishing a Combined Firmware Bundle to `tt-firmware`
+
+Find the `.fwbundle` file and combine it with closed-source firmware versions for older Tenstorrent
```

Comment:
Long term, would it make sense to put the closed source firmware versions as binaries in this repo? That way we can generate releases directly from here. Another option would be to use west's blobs support to pull in the binaries from another URL. Either way, I think it would make sense to get to the point where we can create finalized FW bundles directly from this repo.

Reply:
I agree, it would be nice to release final firmware bundles here for all products, but our official releases are still published on tt-firmware.

Ideally, long term, we should have open source firmware for older products in this repo as well.

Short term, a bit of automation and maybe a deploy hook. Our older FW does not get updated very frequently, so it shouldn't be terrible to manually bundle them either.